realm/realm-library/src/main/java/io/realm/RealmQuery.java

```diff
@@ -1659,86 +1659,10 @@ public long count() {
     }
 
     /**
-     * Finds all objects that fulfill the query conditions and sorted by specific field name.
-     * This method is only available from a Looper thread.
-     *
-     * @return immediately an empty {@link RealmResults}. Users need to register a listener
-     * {@link io.realm.RealmResults#addChangeListener(RealmChangeListener)} to be notified when the query completes.
-     * @see io.realm.RealmResults
+     * @deprecated use {@link #findAll()} instead.
```

Comment:
Yes, so basically the queries will like those in realm-cocoa -- 
1. by default, any queries will be async and executed in the notifier thread. 
2. Any access to a unloaded `RealmResults` will run the query immediately in current thread.
3. Query will be executed again in the notifier thread when the db changes.
4. Handover is handled by the object store automatically.